docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,406 @@
+- we definitely shouldn't configure it both on sidecar and egress, that way we'll have squared retries
+
+##### MeshTimeout
+
+###### Old ExternalService behaviour
+
+Both Timeout and MeshTimeout are configured on the sidecar.
+
+###### New behaviour
+
+If on egress:
+- we would have to match the timeouts from the sidecar not to have a clash (that server times out before client)
+- I don't see any advantages of having this on egress but nothing rules it out
+
+##### MeshTrace
+
+###### Old ExternalService behaviour
+
+Both TrafficTrace and MeshTrace do not support egress.
+
+###### New behaviour
+
+I think it makes sense both on sidecar and egress.
+If we're troubleshooting something it would be great to see spans from sidecar going to ExternalService.
+
+##### MeshTrafficPermission
+
+###### Old ExternalService behaviour
+
+Both MeshTrafficPermission and TrafficPermission only works with egress.
+
+###### New behaviour
+
+It is already implemented on egress for MeshTrafficPermission.
+
+If on the sidecar:
+- could you circumvent this limitation by trying to directly call through egress?
+
+##### MeshLoadBalancingStrategy
+
+###### Old ExternalService behaviour
+
+There is no old behaviour because there was no concept of Endpoints in External Service
+
+###### New behaviour
+
+Only the `loadBalancer` field makes sense in this regard.
+
+If on sidecar:
+- if you don't need/want egress it still makes sense to pick load balancing mechanism
+
+If on egress:
+- makes the same sense on as on sidecar
+
+##### MeshTCPRoute and MeshHTTPRoute
+
+###### Old ExternalService behaviour
+
+It worked for both egress and sidecar.
+
+###### New behaviour
+
+I don't see why we would change that behaviour.
+
+#### For outbound policies on the sidecar, for inbound policies on egress
+
+It naturally fits our targetRef mechanism.
+It does not make sense to apply inbound policies on the sidecar since there is no inbound for it.
+Egress plays a role of a "sidecar" that is inside the mesh, and we can target it's inbound.
+Configuring outbound policies on the sidecar avoids the problem with different clients influencing each other (see [MeshCircuitBreaker](#meshcircuitbreaker)).
+
+##### How to configure it
+
+###### From only policy (policies that do not have to)
+
+Policies:
+- MeshTrafficPermission
+- MeshRateLimit
+- MeshGlobalRateLimit
+- MeshFaultInjection
+
+```yaml
+kind: MeshTrafficPermission
+spec:
+  targetRef:
+    kind: MeshExternalService
+    name: httpbin
+    proxyType: ["egress"]
+  from:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      default:
+        action: Allow
+    - targetRef:
+        kind: Mesh
+      default:
+        action: Deny
+```
+
+###### `To` policy on sidecar
+
+Policies:
+- MeshRetry
+- MeshHTTPRoute
+- MeshTCPRoute
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshRetry
+metadata:
+  name: backend
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: ["sidecar"]
```

Comment:
why should this be necessary?

Reply:
I think it's easier for the user to understand where the policy is applied. I think there are a two options:
- explicit (always state `proxyType`)
- implicit (based on `kind: MeshExternalService` and policy type we implicitly place the configuration on the appropriate component)

I think implicit can get us in trouble fast. Currently "[Empty list: Defaults to targeting all data plane proxies](https://kuma.io/docs/2.8.x/policies/targetref/#understanding-targetref-policies)" so that would mean without `proxyType` we would place the configuration on both `sidecar` and `egress` - that's not good.

On the other hand maybe it's fine to be implicit for MES and we just document it.

Reply:
We decide to configure things implicit